Kros.KORM/src/Kros.KORM/Metadata/ConventionModelMapper.cs

```diff
@@ -265,27 +267,74 @@ private string GetName(ICustomAttributeProvider attributeProvider)
 
         private static IEnumerable<ColumnInfo> OnMapPrimaryKey(TableInfo tableInfo)
         {
-            var ret = new List<ColumnInfo>();
+            ColumnInfo pkByConvention = null;
+            var pkByAttributes = new List<(ColumnInfo Column, KeyAttribute Attribute)>();
 
             foreach (var column in tableInfo.Columns)
             {
                 var attributes = column.PropertyInfo.GetCustomAttributes(typeof(KeyAttribute), true);
                 if (attributes.Length == 1)
                 {
-                    column.AutoIncrementMethodType = (attributes[0] as KeyAttribute).AutoIncrementMethodType;
                     column.IsPrimaryKey = true;
-                    ret.Add(column);
+                    pkByAttributes.Add((column, attributes[0] as KeyAttribute));
                 }
-                else if (column.Name.Equals(ID_NAME, StringComparison.CurrentCultureIgnoreCase))
+                else if (column.Name.Equals(ConventionalPrimaryKeyName, StringComparison.OrdinalIgnoreCase))
                 {
-                    column.IsPrimaryKey = true;
-                    ret.Add(column);
+                    pkByConvention = column;
+                }
+            }
+
+            var ret = new List<ColumnInfo>();
+            if (pkByAttributes.Count == 1)
+            {
+                ret.Add(pkByAttributes[0].Column);
+                ret[0].AutoIncrementMethodType = pkByAttributes[0].Attribute.AutoIncrementMethodType;
+            }
+            else if (pkByAttributes.Count > 1)
+            {
+                CheckPrimaryKeyColumns(pkByAttributes, tableInfo.Name);
+                ret.AddRange(pkByAttributes.OrderBy(item => item.Attribute.Order).Select(item => item.Column));
+                for (int i = 0; i < ret.Count; i++)
+                {
+                    ret[i].PrimaryKeyOrder = i;
                 }
             }
+            else if (pkByConvention != null)
+            {
+                pkByConvention.IsPrimaryKey = true;
+                ret.Add(pkByConvention);
+            }
 
             return ret;
         }
 
+        private static void CheckPrimaryKeyColumns(
+            List<(ColumnInfo Column, KeyAttribute Attribute)> pkAttributes,
+            string tableName)
+        {
+            var uniqueOrders = new HashSet<int>(pkAttributes.Select(item => item.Attribute.Order));
+            if (uniqueOrders.Count != pkAttributes.Count)
```

Comment:
You can use something like this: 
```suggestion
            if (pkAttributes.Select(item => item.Attribute.Order).Distinct().Count() != pkAttributes.Count)
```